lib/std/json/static.zig

```diff
@@ -305,7 +318,7 @@ fn internalParse(
             var fields_seen = [_]bool{false} ** structInfo.fields.len;
 
             while (true) {
-                var name_token: ?Token = try source.nextAllocMax(allocator, .alloc_if_needed, options.max_value_len.?);
+                var name_token: ?Token = try source.nextAllocMax(allocator, options.allocate, options.max_value_len.?);
```

Comment:
Like the union case, but the `freeAllocated` call is already present in the for loop `else` branch.

Reply:
Addressed, with additional `freeAllocated` calls added to the `switch` as described in the above comment.